includes/senders/SenderFactory.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+interface WPNotify_SenderFactory {
+
+	/**
+	 * Create a new instance of notification sender
+	 *
+	 * @param string $name
+	 *
+	 * @return WPNotify_Sender
+	 */
+	public function create( string $name ): WPNotify_Sender;
```

Comment:
I'm not sure if we should use static type hinting or not.

What would be the benefits in your opinion?

Reply:
yep, fixed it by removing it